generator/sbpg/targets/resources/sbp_message_traits_template.h

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (C) 2019 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef SBP_CPP_MESSAGE_TRAITS_H
+#define SBP_CPP_MESSAGE_TRAITS_H
+
+((*- for i in includes *))
+#include <libsbp/(((i))).h>
+((*- endfor *))
+
+namespace sbp {
+
+template<typename>
+struct MessageTraits { };
```

Comment:
is it allowed to have someone manually specify `MessageTraits<double> a;`? If not than static assure to make sure it never gets called unless specialized

Reply:
I'll remove the definition of this class so it's impossible to use it without specializing it. There are a couple of more complicated ways to prevent their use but this seems the most strait forward approach.